scripts/safe_template_linter.py

```diff
@@ -2495,28 +2518,43 @@ def _process_file(full_path, template_linters, options, summary_results, out):
         results.print_results(options, summary_results, out)
 
 
-def _process_current_walk(current_walk, template_linters, options, summary_results, out):
+def _process_os_dir(directory, files, template_linters, options, summary_results, out):
     """
     For each linter, lints all the files in the current os walk.  This means
     finding and printing violations.
 
     Arguments:
-        current_walk: A walk returned by os.walk().
+        directory: Directory being linted.
+        files: All files in the directory to be linted.
         template_linters: A list of linting objects.
         options: A list of the options.
         summary_results: A SummaryResults with a summary of the violations.
         out: output file
 
     """
-    num_violations = 0
-    walk_directory = os.path.normpath(current_walk[0])
-    walk_files = current_walk[2]
-    for walk_file in walk_files:
-        full_path = os.path.join(walk_directory, walk_file)
+    for file in sorted(files, key=lambda s: s.lower()):
+        full_path = os.path.join(directory, file)
         _process_file(full_path, template_linters, options, summary_results, out)
 
 
-def _process_os_walk(starting_dir, template_linters, options, summary_results, out):
+def _skip_os_dir(directory):
+    """
+    Determines if a directory can be skipped for all linters.  Allows the linter
+    to save time by not sorting and linting directories that will never be
+    linted.
+
+    Arguments:
+        directory: The directory to be linted.
+
+    Returns:
+        True if the directory should be skipped, and False otherwise.
+
+    """
+    base_linter = BaseLinter()
+    return base_linter._is_skip_dir(base_linter._skip_dirs, directory)
```

Comment:
This function is here to get access to BaseLinter._is_skip_dir, but that method doesn't need to be a method at all anyway (it doesn't use self).  (btw: _is_skip_dir should use `re.escape` before formatting a string into a regex, or "foo/xgit/bar" will be skipped).